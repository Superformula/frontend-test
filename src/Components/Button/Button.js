import React from 'react';
import PropTypes from 'prop-types';
import "./Button.scss";

function Button(props) {
  return (
    <button className={props.size}>
      {props.children}
    </button>
  );
}

Button.propTypes = {
  size: PropTypes.string
};

export default Button;