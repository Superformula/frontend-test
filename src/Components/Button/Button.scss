// import for color variables
@import '../../styles/index';

.button {
  @include font-regular;
  background-color: $midnightblue;
  border-color: $midnightblue;
  border-radius: 2px;
  color: $white;
  font-size: 14px;
  letter-spacing: 1px;
  line-height: 16px;
  max-width: 420px;
  padding: 13px 0;
  text-align: center;
  text-transform: uppercase;
  width: 100%;
  
  &.inverted {
    background-color: transparent;
    border: 1px solid $midnightblue;
    color: $midnightblue;
  }
}

