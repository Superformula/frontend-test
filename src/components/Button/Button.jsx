import PropTypes from 'prop-types';
import styled from 'styled-components';
import { variant } from 'styled-system';

const buttonStyle = variant({
  key: 'buttons',
});

const Button = styled.button`
  ${buttonStyle}
  display: inline-flex;
  justify-content: center;
  width: ${props => props.block ? '100%' : 'auto'}
  padding: 12px 24px;
  outline: 0;
  font-size: ${props => props.small ? props.theme.fontSizes[0] : props.theme.fontSizes[1]};
  font-stretch:normal;
  text-transform: uppercase;
  border-radius: 2px;
  transition: all cubic-bezier(0.4, 0.0, 0.2, 1) 0.25s;
  cursor: pointer;
  &:disabled {
    cursor: default;
  }
  &:hover:not(:disabled) {
    transform: translateY(2px);
  }
`;

Button.defaultProps = {
  secondary: false,
  block: false,
};

Button.propTypes = {
  secondary: PropTypes.bool,
  block: PropTypes.bool,
};

export default Button;
