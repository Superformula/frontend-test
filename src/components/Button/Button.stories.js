import React from 'react';
import { Button } from './';

const Template = args => <Button {...args}>Click me!</Button>;

export const Default = Template.bind({});
Default.args = { disabled: false, $secondary: false, $fill: false };

export default {
  title: 'Button',
  component: Button,
  argTypes: { onClick: { action: 'clicked' } },
};
