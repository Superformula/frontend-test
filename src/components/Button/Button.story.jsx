import React from 'react';
import { storiesOf } from '@storybook/react';
import {
  Button,
  GridItem,
  Heading,
  SubHeading,
} from '..';

storiesOf('Button', module)
  .add('default', () => (
    <GridItem p={2}>
      <Heading>Default Button</Heading>
      <SubHeading>Large</SubHeading>
      <GridItem py={2}>
        <Button variant="primary">Default</Button>
      </GridItem>
      <GridItem py={2}>
        <Button variant="primary" disabled>Disabled</Button>
      </GridItem>
      <GridItem py={2}>
        <Button variant="primary" block>Block</Button>
      </GridItem>
      <GridItem py={2}>
        <SubHeading>Small</SubHeading>
        <Button variant="primary" small>Default</Button>
      </GridItem>
      <GridItem py={2}>
        <Button variant="primary" small disabled>Disabled</Button>
      </GridItem>
      <GridItem py={2}>
        <Button variant="primary" small block>Block</Button>
      </GridItem>
    </GridItem>
  ))
  .add('secondary', () => (
    <GridItem p={2}>
      <Heading>Secondary Button</Heading>
      <SubHeading>Large</SubHeading>
      <GridItem py={2}>
        <Button variant="secondary">Secondary</Button>
      </GridItem>
      <GridItem py={2}>
        <Button variant="secondary" disabled>Disabled</Button>
      </GridItem>
      <GridItem py={2}>
        <Button variant="secondary" block>Block</Button>
      </GridItem>
      <GridItem py={2}>
        <SubHeading>Small</SubHeading>
        <Button variant="secondary" small>Secondary</Button>
      </GridItem>
      <GridItem py={2}>
        <Button variant="secondary" small disabled>Disabled</Button>
      </GridItem>
      <GridItem py={2}>
        <Button variant="secondary" small block>Block</Button>
      </GridItem>
    </GridItem>
  ));
