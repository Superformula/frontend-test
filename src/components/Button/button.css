.button {
  border-radius: 1px;
  border-color: var(--blue);
  padding-top: var(--medium-padding);
  padding-bottom: var(--medium-padding);
  color: var(--blue);
}

.button:hover {
  cursor: pointer;
  color: var(--white);
  background-color: var(--blue);
}

.filled {
  color: var(--white);
  background-color: var(--blue);
}

.filled:hover {
  color: var(--blue);
  background-color: var(--white);
}
