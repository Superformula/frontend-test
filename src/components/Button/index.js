import React from 'react';
import classnames from 'classnames';
import style from './button.css';

const Button = ({
  label,
  filled,
  onClick,
  className
}) => (
  <button
    onClick={onClick}
    className={classnames(className, {
      [style.button]: true,
      [style.filled]: Boolean(filled)
    })}
  >
    {label}
  </button>
)

export default Button;
