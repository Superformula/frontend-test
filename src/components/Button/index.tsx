import styled from "@emotion/styled";

// TODO probably this should expand
// the native button props
export interface IProps {
  fullWidth?: boolean;
  inverse?: boolean;
  disabled?: boolean;
  type?: string;
  onClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => any;
  children: React.ReactNode;
}

const Button = styled.button<IProps>(
  ({ theme, fullWidth, inverse, disabled }) => {
    // TODO extract this into a function that
    // can be super easily be unit tested
    let bgColor = theme.colors.primary;
    let fgColor = theme.colors.white;
    let borderColor = theme.colors.primary;

    if (disabled) {
      bgColor = theme.colors.grey100;
      fgColor = theme.colors.grey200;
      borderColor = theme.colors.grey100;
    }

    if (inverse) {
      bgColor = theme.colors.white;
      fgColor = theme.colors.primary;
      borderColor = theme.colors.primary;

      if (disabled) {
        bgColor = theme.colors.white;
        fgColor = theme.colors.grey200;
        borderColor = theme.colors.grey100;
      }
    }

    return `
    background-color: ${bgColor};
    color: ${fgColor};
    border: 1px solid ${borderColor};
    border-radius: ${theme.borderRadius.small};
    padding: 16px;
    text-align: center;
    font-size: ${theme.fontSize.x400};
    text-transform: uppercase;
    cursor: pointer;
    ${fullWidth ? `width: 100%;` : ``}

    &:hover {
      background-color: ${theme.colors.primary}cc;
    }
  `;
  }
);

export default Button;
