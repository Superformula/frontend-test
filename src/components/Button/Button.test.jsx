import React from 'react';
import { mountWithTheme } from '../../helpers/tests/withTheme';
import Button from './Button';

describe('Button Component', () => {
  let component;
  it('renders a primary button correctly', () => {
    component = mountWithTheme(<Button variant="primary">Primary</Button>).find('button');
    expect(component).toMatchSnapshot();
  });
  it('renders a secondary button correctly', () => {
    component = mountWithTheme(<Button variant="secondary">Secondary</Button>).find('button');
    expect(component).toMatchSnapshot();
  });
  it('renders a disabled button correctly', () => {
    component = mountWithTheme(<Button variant="primary" disabled>Disabled</Button>).find('button');
    expect(component).toMatchSnapshot();
  });
});
