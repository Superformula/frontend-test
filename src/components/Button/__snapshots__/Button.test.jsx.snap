// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders a disabled button correctly 1`] = `
<button
  className="sc-bdVaJa crknUg"
  disabled={true}
>
  Disabled
</button>
`;

exports[`Button Component renders a primary button correctly 1`] = `
<button
  className="sc-bdVaJa crknUg"
>
  Primary
</button>
`;

exports[`Button Component renders a secondary button correctly 1`] = `
<button
  className="sc-bdVaJa dqbvwZ"
>
  Secondary
</button>
`;
